File: customer-java-resttemplate/src/main/java/com/prez/lib/health/info/EnvInfosContributor.java
package com.prez.lib.health.info;

import java.util.Map;
import java.util.function.Function;
import java.util.regex.Pattern;
import java.util.stream.Collectors;
import javax.annotation.PostConstruct;
import org.springframework.boot.actuate.info.Info;
import org.springframework.boot.actuate.info.InfoContributor;
import org.springframework.core.env.ConfigurableEnvironment;
import org.springframework.core.env.MapPropertySource;

public class EnvInfosContributor implements InfoContributor {


    private static final String OBFUSCATED = "******";
    private final Pattern sensiblePropsNamePattern;
    private ConfigurableEnvironment environment;
    private Map<String, Object> propertyMap;

    public EnvInfosContributor(ConfigurableEnvironment environment, String sensiblePropsNameRegex) {
        this.environment = environment;
        this.sensiblePropsNamePattern = Pattern.compile(sensiblePropsNameRegex);
    }

    @PostConstruct
    void buildPropertyMap() {
        propertyMap = filterSystemProperties();//to keep Map State
    }

    @Override
    public void contribute(Info.Builder builder) {
        builder.withDetail("properties", propertyMap);
    }

    private Map<String, Object> filterSystemProperties() {
        return environment.getPropertySources().stream()
                .filter(MapPropertySource.class::isInstance)
                .filter(propertySource -> !"systemProperties".equals(propertySource.getName()) && !"systemEnvironment".equals(propertySource.getName()))
                .map(MapPropertySource.class::cast)
                .flatMap(mapPropertySource -> mapPropertySource.getSource().keySet().stream())
                .distinct()
                .collect(Collectors.toMap(Function.identity(), this::obfuscateSensiblesProps));
    }

    private String obfuscateSensiblesProps(String key) {
        if (sensiblePropsNamePattern.matcher(key.toLowerCase()).matches()) {
            return OBFUSCATED;
        } else {
            return environment.getProperty(key);
        }
    }

}

File: customer-kotlin-coroutines/src/main/kotlin/com/prez/lib/health/info/EnvInfosContributor.kt
package com.prez.lib.health.info

import org.springframework.boot.actuate.info.Info
import org.springframework.boot.actuate.info.InfoContributor
import org.springframework.core.env.ConfigurableEnvironment
import org.springframework.core.env.MapPropertySource
import java.util.regex.Pattern
import javax.annotation.PostConstruct

class EnvInfosContributor(private val environment: ConfigurableEnvironment, sensiblePropsNameRegex: String) :
  InfoContributor {

  private val sensiblePropsNamePattern: Pattern = Pattern.compile(sensiblePropsNameRegex)
  private var propertyMap: Map<String, Any>? = null

  companion object {
    private const val OBFUSCATED = "******"
  }

  @PostConstruct
  fun buildPropertyMap() {
    propertyMap = filterSystemProperties() //to keep Map State
  }

  override fun contribute(builder: Info.Builder) {
    builder.withDetail("properties", propertyMap)
  }

  private fun filterSystemProperties(): Map<String, Any> {
    return environment.propertySources
      .filterIsInstance(MapPropertySource::class.java)
      .filter { "systemProperties" != it.name && "systemEnvironment" != it.name }
      .map { MapPropertySource::class.java.cast(it) }
      .flatMap { it.source.keys }
      .distinct()
      .associateBy({ it }) { key: String -> obfuscateSensiblesProps(key) }
    //.collect(Collectors.toMap(Function.identity(), { key: String -> obfuscateSensiblesProps(key) }))
}

private fun obfuscateSensiblesProps(key: String): String {
  return if (sensiblePropsNamePattern.matcher(key.toLowerCase()).matches()) OBFUSCATED else environment.getProperty(key)
}
}


